azurerm/internal/services/network/tests/vpn_gateway_connection_resource_test.go

```diff
@@ -0,0 +1,396 @@
+  }
+
+  vpn_link_connection {
+    name             = "link3"
+    vpn_site_link_id = azurerm_vpn_site.test.link[1].id
+  }
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMVpnGatewayConnection_customRouteTableUpdate(data acceptance.TestData) string {
+	template := testAccAzureRMVpnGatewayConnection_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_virtual_hub_route_table" "test" {
+  name           = "acctest-RouteTable-%[2]d"
+  virtual_hub_id = azurerm_virtual_hub.test.id
+}
+
+resource "azurerm_virtual_hub_route_table" "test2" {
+  name           = "acctest-RouteTable-%[2]d-2"
+  virtual_hub_id = azurerm_virtual_hub.test.id
+}
+
+resource "azurerm_vpn_gateway_connection" "test" {
+  name               = "acctest-VpnGwConn-%[2]d"
+  vpn_gateway_id     = azurerm_vpn_gateway.test.id
+  remote_vpn_site_id = azurerm_vpn_site.test.id
+  routing_configuration {
+    associated_route_table  = azurerm_virtual_hub_route_table.test2.id
+    propagated_route_tables = [azurerm_virtual_hub_route_table.test2.id]
+  }
+  vpn_link_connection {
+    name             = "link1"
+    vpn_site_link_id = azurerm_vpn_site.test.link[0].id
+    ipsec_policy {
+      sa_lifetime_sec            = 300
+      sa_data_size_kb            = 1024
+      ipsec_encryption_algorithm = "AES256"
+      ipsec_integrity_algorithm  = "SHA256"
+      ike_encryption_algorithm   = "AES128"
+      ike_integrity_algorithm    = "SHA256"
+      dh_group                   = "DHGroup14"
+      pfs_group                  = "PFS14"
+    }
+    bandwidth_mbps                    = 30
+    protocol                          = "IKEv2"
+    ratelimit_enabled                 = true
+    route_weight                      = 2
+    shared_key                        = "secret"
+    use_local_azure_ip_address        = true
+    use_policy_based_traffic_selector = true
+  }
+
+  vpn_link_connection {
+    name             = "link3"
+    vpn_site_link_id = azurerm_vpn_site.test.link[1].id
+  }
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMVpnGatewayConnection_requiresImport(data acceptance.TestData) string {
+	template := testAccAzureRMVpnGatewayConnection_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_vpn_gateway_connection" "import" {
+  name               = azurerm_vpn_gateway_connection.test.name
+  vpn_gateway_id     = azurerm_vpn_gateway_connection.test.vpn_gateway_id
+  remote_vpn_site_id = azurerm_vpn_gateway_connection.test.remote_vpn_site_id
+  dynamic "vpn_link_connection" {
+    for_each = azurerm_vpn_gateway_connection.test.vpn_link_connection
+    iterator = v
+    content {
+      name             = v.value["name"]
+      vpn_site_link_id = v.value["vpn_site_link_id"]
+    }
+  }
+}
+`, template)
+}
+
+func testAccAzureRMVpnGatewayConnection_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-rg-%[1]d"
```

Comment:
we always use acctestRG to ensure capitals work correctly and include the service
```suggestion
  name     = "acctestRG-vpn-%[1]d"
```